www/pages/api.php

```diff
@@ -215,12 +225,36 @@ function($release) {
 
 	// Capabilities request.
 	case 'c':
+		//get categories
 		$category = new Category(['Settings' => $page->settings]);
-		$page->smarty->assign('parentcatlist', $category->getForMenu());
-		header('Content-type: text/xml');
-		echo $page->smarty->fetch('apicaps.tpl');
-		break;
+		$cats = $category->getForMenu();
+
+		//insert cats into template variable
+		$page->smarty->assign('parentcatlist', $cats);
 
+		if ($outputXML) {
+			//use apicaps.tpl if xml is requested
+			header('Content-type: text/xml');
+			echo $page->smarty->fetch('apicaps.tpl');
+		}
+		 else {
+			 //otherwise construct array of capabilities and categories
+
+			 //get capabilities
+			 $capability = new Capabilities(['Settings' => $page->settings]);
+			 $caps = $capability->getForMenu();
+
+			 //create array
+			 $retJson = array();
+			 //write capability information
+			 $retJson = $caps;
+			 //append category information
+			 $retJson["categories"] = $cats;
+			//use json_encode
+			 header('Content-type: application/json');
+			 echo json_encode($retJson);
```

Comment:
Bad spacing, also many of these lines can be combined:

`header('Content-type: application/json');`
`echo json_encode(['categories' => (new Capabilities(['Settings' => $page->settings]))->getForMenu()]);`